### PR TITLE
refac: Buildblocks infrastructure

### PR DESCRIPTION
**Pull request by lasrinnil**
## Description
To circumvent a circular reference from the `Infrastructure` project which holds our `UnitOfWork` that knows each `*.Infrastructure` to handle processing for each `DbContext` in a single transaction, and the fact that the shared infrastructure was placed in the same project. 

I have moved the `SQLConnetionFactory` and the `MessageBus` infrastructure stuff out and into a `BuildingBlocks.Infrastructure` project - I have been inspired to this approach from this repo: https://github.com/kgrzybek/modular-monolith-with-ddd

In the future "Common" will be empty (hopefully) and its content moved to corresponding `BuildingBlock` projects.
image

### CODE REVIEW

File: source/ArchivedMessages.Infrastructure/ArchivedMessages.Infrastructure.csproj [hunk 1]
```diff
@@ -13,6 +13,7 @@
 
   <ItemGroup>
     <ProjectReference Include="..\ArchivedMessages.Interfaces\ArchivedMessages.Interfaces.csproj" />
```
Comment (MadsDue) [inline source/ArchivedMessages.Infrastructure/ArchivedMessages.Infrastructure.csproj:15]: Are Interfaces and Contracts representing the same thing or?

Comment (MadsDue) [inline source/ArchivedMessages.Infrastructure/ArchivedMessages.Infrastructure.csproj:15]: Should the Infrastructure layer know the Interfaces layer? should it not be the application.

**Review by MadsDue — APPROVED** [review]

File: source/ArchivedMessages.Infrastructure/ArchivedMessages.Infrastructure.csproj (hunk 1 above)
Comment (lasrinnil) [reply source/ArchivedMessages.Infrastructure/ArchivedMessages.Infrastructure.csproj:15]: I'll do a refactor that will remove this dependency

Comment (lasrinnil) [reply source/ArchivedMessages.Infrastructure/ArchivedMessages.Infrastructure.csproj:15]: In another PR

**Pull request merged**